CONTRIBUTING.md

```diff
@@ -49,134 +37,181 @@ and let us know if it's not up-to-date (even better, submit a PR with your  corr
+docker$ exit
 ```
 
-Alternatively, the [Full Docker build](#full-docker-build) workflow can be used so
-that you don't have to worry about installing all those tools on your local machine
-and instead only depend on having a working Docker install.
+The `dist/` folder within each module contains the packaged up language artifacts.
 
-## Pull Requests
+## Docker Build (Alternative)
 
-### Pull Request Checklist
-
-* [ ] Testing
-  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
-  - __CLI change?:__ coordinate update of integration tests with team
-  - __cdk-init template change?:__ coordinated update of integration tests with team
-* [ ] Docs
-  - __jsdocs__: All public APIs documented
-  - __README__: README and/or documentation topic updated
-  - __Design__: For significant features, design document added to `design` folder
-* [ ] Title and Description
-  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
-  - __Title__: use lower-case and doesn't end with a period
-  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
-  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
-* [ ] Sensitive Modules (requires 2 PR approvers)
-  - IAM Policy Document (in @aws-cdk/aws-iam)
-  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
-  - Grant APIs (only if not based on official documentation with a reference)
-
----
-
-### Step 1: Open Issue
-
-If there isn't one already, open an issue describing what you intend to contribute. It's useful to communicate in
-advance because if someone is already working in this space, it may be worth collaborating with them
-instead of duplicating the effort.
+Build the docker image:
 
-### Step 2: Design (optional)
+```console
+$ docker build -t aws-cdk .
+```
 
-In some cases, it is useful to seek feedback by iterating on a design document. This is useful
-when you plan a big change or feature, or you want advice on what would be the best path forward.
+This allows you to run the CDK in a CDK-compatible directory with a command like:
+
+```console
+$ docker run -v $(pwd):/app -w /app aws-cdk <CDK ARGS>
+```
+
+## Gitpod (Alternative)
+
+You may also set up your local development environment using [Gitpod](http://gitpod.io) -
+a service that allows you to spin up an in-browser Visual Studio Code-compatible editor,
+with everything set up and ready to go for CDK development.
+Just click the button below to create your private workspace:
+
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/aws/aws-cdk)
+
+This will start a new Gitpod workspace, and immediately kick off a build of the CDK code.
+Once it's done (it takes around an hour, unfortunately), you can work on any package that you want to modify,
+as described in ['Quick Iteration'](#quick-iteration) below.
```

Comment:
Fortunately, we now use pre-builds, so the one-hour wait time is no more! We can remove that part.